### PR TITLE
`FlowExecution.getOwner` left null after `WorkflowRun.reload`

### PR DESCRIPTION
**Pull request by jglick**
Would not normally matter I think, but can cause confusing error messages from `CpsFlowExecution` which normally assumes that its `owner` is non-null at almost all times; in particular `toString` then makes no mention of which build is having issues.

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java [hunk 1]
```diff
@@ -553,6 +553,10 @@ private String key() {
 
         // super.reload() forces result to be FAILURE, so working around that
         new XmlFile(XSTREAM,new File(getRootDir(),"build.xml")).unmarshal(this);
+        var _execution = execution;
```
Comment (dwnusbaum) [inline src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java:556]: For what it's worth, my first thought is to instead null out `execution` and reset `executionLoaded` and let them be reloaded by `getExecution` only if needed.

If you prefer to keep the current approach, I would add a `synchronized (this)` block to match `WorkflowRun.getExecution` (unless perhaps typical callers already acquire the monitor for `WorkflowRun` in this case).

Comment (jglick) [reply src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java:556]: > null out `execution` and reset `executionLoaded`

:thinking: that may work better. Will check it when I get a moment.

Comment (jglick) [reply src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java:556]: It does not work; `getExecution` simply returns null in that case.

**Comment by jglick** [comment]
This seems to contradict the `lazyLoadExecution` test from #93 :eyes:

**Comment by jglick** [comment]
`WorkflowRunTest.culprits` still sometimes hanging, not sure why.

**Review by dwnusbaum — APPROVED** [review]

**Review by dwnusbaum — APPROVED** [review]

**Pull request merged**